Adds the hypnotic flash to the uplink

XDTM opened the pull request:
:cl: XDTM
add: Added the Hypnotic Flash to the uplink for 7 TC.
add: The Hypnotic Flash temporarily confuses and pacifies those it's used on.
add: If the victim is in a mentally vulnerable state (hallucinating, insane, reduced mental activity) they will instead fall into a trance, and will be hypnotized by the next words they hear.
/:cl:

Controlled hypnosis for traitors. Usual caveats apply: Mindshield prevents the trance effect (but not the confusion/pacification) and it can still deconvert from hypnosis; a new hypnosis will override the previous one; and most importantly for any ambiguity the interpretation is mostly up to the victim, so it's not a 100% sure conversion (i'm _hoping_ people won't use this clause to just ignore it).

XDTM committed:
Adds the hypnotic flash as a role-restricted traitor tool

XDTM committed:
fix

SouDescolado commented:
Could this be made so when you flash someone you already say the words the victim should follow? So it will get a special alert in the screen saying "You feel compelled to follow: [thecommand]" instead of trying to follow a random command, like.. idk, "Honk"

Also, if I've read the code right, if a pacifist get hypnotized, s/he won't lose pacifism, right?

QualityVan commented on code/modules/assembly/flash.dm:
```diff
@@ -311,3 +312,60 @@
+/obj/item/assembly/flash/hypnotic/burn_out()
+	return
+
+/obj/item/assembly/flash/hypnotic/proc/check_vulnerable(mob/living/carbon/C)
```
Is this leftovers from WIP? It's never called and it looks like its role is taken by carbon's hypnosis_vulnerable()

XDTM replied:
Oh i forgot to delete tgat yeah

XDTM commented:
If you want you can pretype an order and send it after flashing to be safe, but i'm leaving a window open for shenanigans (i.e. radio messages getting there first).
It can also be used as a 20 seconds stun under the right conditions.

ShizCalev commented on code/datums/status_effects/debuffs.dm:
```diff
@@ -83,6 +83,24 @@
+/datum/status_effect/pacify/on_creation(mob/living/new_owner, set_duration)
+	if(isnum(set_duration))
+		duration = set_duration
+	. = ..()	
```
return ..()

XDTM committed:
fyxes

davethwave commented:
This will be fun

davethwave commented:
Also a suggestion, perhaps make it work on people who have a certain amount of brain damage?

davethwave commented:
Though if a price raise is not going to happen then extend the cooldown? Two seconds is not enough in my opinion.

XDTM commented:
I set the price kinda low since it's role restricted, has limitations, and it requires additional conditions to work in the first place. It can be raised if it's too cheap, though.

I considered checking for brain damage, but the way brain damage works is more of a predisposition to traumas than an actual reduction in cognitive abilities, so i only checked for related trauma symptoms (i.e. hallucinations or the dumb trait). Being paralyzed or having twitchy arms shouldn't cause psychological vulnerability, after all.

XDTM commented:
About the cooldown; i've set it to two seconds so you can't spam-AoE it since it doesn't burn out, but otherwise i don't think it needs a cooldown mechanic by itself; having a controlled hypnosis already requires you to restrain and remove radios from the target and get into a secluded room, then dose them with mindbreaker, then perform the hypnosis being careful of the wording. Should be enough hurdles without a cooldown.

davethwave commented:
OK, that makes sense. However the cost if the flash may be the deciding point of if this gets merged or not. However since you can get a power sink for the same price I'm not very worried.

obscolene commented:
literal fetish content

subject217 commented:
time to add vore as a traitor item

Vidiveni commented:
This is a sensible use of the mood mechanic. You could try to tank someone's mood in order to hypnotize them. The item will also be fun after SM delams. 

Might have to think of a policy about cheating the '1 word rule' by saying "killLexiaBlackatallcosts" and such.

XDTM commented:
What one word rule? You could hypnotize someone with wgw if you want to, the only downside is that they won't get the fancy pulsing text since they will never hear it in normal conversation

ATH1909 commented:
I feel like having the dumb trait should make you _immune_ to the flash, not vulnerable to it. Y'know, the whole "too dumb to hypnotize" trope.

If it works based on flashing them, how are they seeing the flash while asleep (can sleeping people even hear commands while asleep currently?)? Does it work on blind people or people with sunglasses?

Make it work on drunk people as well.

What if you made it hypnotize confused people as well, but increased the cooldown so you can't just flash someone twice in a row to hypnotize them?

Would adding a HUD that displayed what people could be hypnotized by your flash be too much? Maybe on an advanced flash? I don't see how you'd be able consistently use it on insane people otherwise (other than by dismembering them or something and waiting?).

Make failed hypnosis attempts slur speech in addition to confuse and pacify victims.

Make the AoE effect hypnotize vulnerable people (but have a long cooldown so it isn't spammed); it could lead to some shenanigans/fun moments.

Maybe make borgs that are built with the hypnotic flash able to use it as a module?

Are you currently able to repeatedly flash someone so that their drowsiness racks up high enough to cause them to fall asleep?

There've gotta be some mind-related conditions/status effects that we're missing...

XDTM commented:
You cannot hypnotize soneone who's asleep due to blindness; i added that check for future sources of hypnosis

XDTM commented:
> > You cannot hypnotize soneone who's asleep due to blindness; i added that check for future sources of hypnosis
> 
> /mob/living/carbon/proc/hypnosis_vulnerable()
> if(has_trait(TRAIT_MINDSHIELD))
> return FALSE
> if(hallucinating())
> return TRUE
> **if(IsSleeping()) return TRUE**
> if(has_trait(TRAIT_DUMB))
> return TRUE
> GET_COMPONENT_FROM(mood, /datum/component/mood, src)
> if(mood)
> if(mood.sanity < SANITY_UNSTABLE)
> return TRUE
> 
> Why is that bolded part in there, then?

Because
> > i added that check for future sources of hypnosis


> Better idea than my "advanced hypnotic flash" idea: Add a type of HUDgoggles for traitors that lets you see mood and sanity instead of health, wanted status, etc..

Maybe not goggles, but it could be a health analyzer/adv. health analyzer function, or perhaps a new psychoanalyzer device (which could take on trauma detection from health analyzers).

> Maybe let the hypnotic flash work on people that have been standing still and in a menu for 30+ seconds (and thus presumably focused on something)?

The only thing i could check for code-wise is if they are afk, and if they are they're gonna miss the hypnosis prompt in any case.

> Also, why is this a role-restricted item? Sure, it's a flash, but I want to hypnotize sad people into being creepily happy all of the time as a clown, convert people to my religion as a chaplain, hypnotize people who get high off of my plants as a botanist, spike drinks with hallucinogens as a bartender, etc.

Fair enough, actually, i might remove the restriction.

XDTM committed:
unrestrict

XDTM commented:
Unrestricted, now everyone can buy it. Increased the cost by one TC.

Shadowflame909 commented:
> 
> 
> Unrestricted, now everyone can buy it. Increased the cost by one TC.
> All things considered, doctors already have the brainwashing disk as an exclusive for pseudoconversion.

I agree with this. Especially since medical brainwashing is easier to use, and allows you to put more concise directives in a timely manner.

This one, you could end up with _;Ai: Hit or miss, I guess they never miss huh?_ as an accidental directive in the 20 second timespan

Cyberboss commented on code/modules/assembly/flash.dm:
```diff
@@ -311,3 +312,56 @@
+		else
+			to_chat(M, "<span class='danger'>[src] fails to blind you!</span>")
+	else
+		if(M.flash_act())
```
`else if ...`

Cyberboss commented on code/modules/assembly/flash.dm:
```diff
@@ -311,3 +312,56 @@
+		else if(user)
+			visible_message("<span class='disarm'>[user] fails to blind [M] with the flash!</span>")
+			to_chat(user, "<span class='warning'>You fail to hypno-flash [M]!</span>")
+			to_chat(M, "<span class='danger'>[user] fails to blind you with the flash!</span>")
```
Is the text from the `visible_message` and second `to_chat` duplicated? If so, add a new proc for them on `/flash`

improvedname commented:
Flash. VIVA LA REV

XDTM committed:
prettier code

XDTM committed:
dent

ExcessiveUseOfCobblestone commented:
If people ignore hypnosis it should become an admin issue. Personally i'd find it similar to a custom AI law, if it doesn't conflict with something that could be justified then it should be followed with room for interpretation.

optimumtact commented:
Implying our players would ignore a directive to let them murder
